File: JavaScript/7js 1/compare.js
/*
|--------------------------------------------------------------------------
| Compare Strings by Sum of Chars
|--------------------------------------------------------------------------
|
*/

function compare(s1, s2) {
   const res = (s) =>
      !s || /[^a-z]/i.test(s)
         ? 0
         : s.split("").reduce((a, b) => a + b[0].toUpperCase().charCodeAt(), 0);
   return res(s1) == res(s2);
}


function compare(s1, s2) {
   if (/^\D+$/gi.test(s1) && /^\D+$/gi.test(s2))
      return (
         s1
            .split("")
            .map((n) => n.toUpperCase().charCodeAt())
            .reduce((a, b) => a + b) ===
         s2
            .split("")
            .map((n) => n.toUpperCase().charCodeAt())
            .reduce((a, b) => a + b)
      );
   return true;
}


function compare(s1, s2) {
   function sum(s) {
      let a = (s || "").toUpperCase().split("");
      return a.every((v) => /[A-Z]/.test(v)) ? a.reduce((r, v) => r + v.codePointAt(0), 0) : 0;
   }
   return sum(s1) === sum(s2);
}


console.log(compare("AD", "BC"), true);
console.log(compare("AD", "DD"), false);
console.log(compare("gf", "FG"), true);
console.log(compare("zz1", ""), true);

File: Python/7py 1/compare.py
"""
|--------------------------------------------------------------------------
| Compare Strings by Sum of Chars
|--------------------------------------------------------------------------
|
"""

def compare(s1,s2):
   res = lambda i: sum(map(ord,i.upper())) if i and i.isalpha() else ''
   return res(s1) == res(s2)


def compare(s1, s2):
   return len(set(sum(map(ord, x.upper() if x and x.isalpha() else '')) for x in [s1, s2])) == 1


def compare(s1,s2):
   a, b =  (sum(ord(c) for c in s.upper()) for s in (s1, s2) if s.isalpha())
   return a == b


print(compare("AD", "BC"), True)
print(compare("AD", "DD"), False)
print(compare("gf", "FG"), True)
print(compare("zz1", ""), True)